R/03-formatGGplot.R

```diff
@@ -22,37 +22,39 @@ formatTitlesOfGGplot <- function(plot, text) {
 
   # AXES ----
   if (any(grepl("Axis", names(text)))) {
-    # set titles
-    axisTitleX <- text[["xAxisTitle"]]
-    axisTitleY <- text[["yAxisTitle"]]
+    plot <- plot %>%
+      setCustomTitle(labFun = xlab, label = text[["xAxisTitle"]][["text"]])
+    plot <- plot %>%
+      setCustomTitle(labFun = ylab, label = text[["yAxisTitle"]][["text"]])
 
-
-    if (axisTitleX[["text"]] != "")
-      plot <- plot + xlab(axisTitleX[["text"]])
-    if (axisTitleY[["text"]] != "")
-      plot <- plot + ylab(axisTitleY[["text"]])
-
-    # apply text formatting
+    # apply text formatting (theme)
     plot <- plot +
       theme(
-        axis.title.x = getElementText(axisTitleX),
-        axis.title.y = getElementText(axisTitleY),
+        axis.title.x = getElementText(text[["xAxisTitle"]]),
+        axis.title.y = getElementText(text[["yAxisTitle"]]),
         axis.text.x = getElementText(text[["xAxisText"]]),
         axis.text.y = getElementText(text[["yAxisText"]])
       )
   }
 
   # PLOT TITLE ----
   if (any(grepl("plot", names(text)))) {
-    plotTitle  <- text[["plotTitle"]]
-    if (plotTitle[["text"]] != "")
-      plot <- plot + ggtitle(label = plotTitle[["text"]])
+    plot <- plot %>%
+      setCustomTitle(labFun = ggtitle, label = text[["plotTitle"]][["text"]])
 
-    plot <- plot + theme(plot.title   = getElementText(plotTitle))
+    # apply text formatting (theme)
+    plot <- plot + theme(plot.title = getElementText(text[["plotTitle"]]))
   }
 
   # LEGEND ----
   if (any(grepl("legend", names(text)))) {
+    plot <- plot %>%
+      setCustomTitle(labFun = labs,
+                     color = text[["legendTitle"]][["text"]],
```

Comment:
Can you explain why `text[["legendTitle"]][["text"]]` for all the arguments here?

Reply:
Currently we have only one input field for a `legend title`. In OsteoBioR (for which this module was developed) we have one grouping variable that we use for all: color, size, fill and shape. Therefor I set the same title for all legends. In essence, we do have only a single legend:
![image](https://github.com/user-attachments/assets/bc005cf0-2410-40cd-b0d1-21902e26356e)
Since currently we do not need more grouping variables, I did not thought about how to implement it more generally.
This would also require to add inputs for more legends. Do you think, we should make a note somewhere (and have an idea where)?

Reply:
Okay sounds good. We can keep it as is and if we should need more grouping variables in the future we can still think about writing it different.